query/sources/impl/src/main/java/org/codice/ddf/admin/sources/commons/utils/SourceValidationUtils.java

```diff
@@ -0,0 +1,98 @@
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.sources.commons.utils;
+
+import static org.codice.ddf.admin.sources.commons.SourceActionCommons.getAllSourceReferences;
+import static org.codice.ddf.admin.sources.commons.SourceMessages.duplicateSourceNameError;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import org.codice.ddf.admin.api.action.Message;
+import org.codice.ddf.admin.common.fields.base.scalar.StringField;
+import org.codice.ddf.admin.configurator.ConfigReader;
+import org.codice.ddf.admin.configurator.ConfiguratorFactory;
+import org.codice.ddf.admin.sources.fields.ServicePid;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.service.ConfiguredService;
+import ddf.catalog.source.Source;
+
+public class SourceValidationUtils {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SourceValidationUtils.class);
+
+    /**
+     * Validates the {@code sourceName} against the existing source names in the system. An empty {@link List} will be returned
+     * if there are no existing source names with with name {@code sourceName}, or a {@code List} containing
+     * {@link Message}s if there are errors.
+     *
+     * @param sourceName          source name to validate
+     * @param configuratorFactory configurator factory for reading FederatedSource service references
+     * @param servicePid          if provided, signifies this is an update on the configuration identified by the pid and if the sourceName
+     *                            matches the servicePid's configuration's sourceName, it is a valid sourceName
+     * @return a {@code List} of {@link Message}s containing a {@link org.codice.ddf.admin.sources.commons.SourceMessages#DUPLICATE_SOURCE_NAME} error, or an empty {@link List}
+     * if there are no duplicate source names found
+     */
+    // TODO: 4/24/17 phuffer -  adding a duplicate name should be valid as long as the Active Binding is different
+    public static List<Message> validateSourceName(StringField sourceName,
+            ConfiguratorFactory configuratorFactory, ServicePid servicePid) {
+        ConfigReader configReader = configuratorFactory.getConfigReader();
+        List<Source> sources = getAllSourceReferences(configuratorFactory);
+
+        if(servicePid != null && servicePid.getValue() != null) {
+            Map<String, Object> existingConfig = configReader.getConfig(servicePid.getValue());
+            if (existingConfig.get("id") != null && existingConfig.get("id")
```

Comment:
This assumes the given service had a service property called "id". There is nothing enforcing this from the api standpoint. There should be a way to pull a service reference using a filter on the pid to call `getId` on the reference's interface instead.